detection_rules/rule_validators.py

```diff
@@ -37,28 +42,39 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
             return
 
         indexes = data.index or []
-        beats_version = meta.beats_version or beats.get_max_version()
-        ecs_versions = meta.ecs_versions or [ecs.get_max_version()]
+        stack_versions = get_stack_schemas(meta.min_stack_version)
```

Comment:
I think we only need to validate against the minimum stack version, right? Do we need to validate against all stack versions?

Assuming there are never breaking changes?

After the 7.x series, we may have to revisit this.

Reply:
that depends if we consider schema changes breaking since beats and ECS change every release. I think it is safer and worth the extra cycles.

Reply:
I see where you're going and you make good points.
I think the main question is determining what "max_stack_version" means...

**How about this:**
Every version between `min_stack_version` and `branch_stack_version` (aka packages.yml) is what's validated?
This way, if you push a 7.13 rule to a 7.15 stack branch, you validate [7.13, 7.14, 7.15] and you don't care about 7.16+.

This will be more portable logic for users as well that fork the repo and are using a stack at a particular point in time.

Reply:
👍 yea that makes sense to me

Reply:
e046660